addons/mass_mailing/models/mass_mailing.py

```diff
@@ -408,230 +407,68 @@ def _get_statistics(self, cr, uid, ids, name, arg, context=None):
                 m.id IN %s
             GROUP BY
                 m.id
-        """, (tuple(ids), ))
-        for row in cr.dictfetchall():
-            results[row.pop('mailing_id')] = row
-            total = row['total'] or 1
+        """, (tuple(self.ids), ))
+        for row in self.env.cr.dictfetchall():
+            total = row.pop('total') or 1
```

Comment:
because the total here is count of statistics and we need total recipients which is assigned to the total field by _compute_total function on line 365.